client/cmd.go

```diff
@@ -147,6 +151,46 @@ func ReadPersistentCommandFlags(clientCtx Context, flagSet *pflag.FlagSet) (Cont
 		}
 	}
 
+	isOffline, _ := flagSet.GetBool(flags.FlagOffline)
+	if !isOffline && clientCtx.ClientConn == nil {
+		grpcURI, _ := flagSet.GetString(flags.FlagGRPCNode)
+		if grpcURI != "" {
+			if !flagSet.Changed(flags.FlagGRPCNode) && rpcURI != "" {
+				if uri, err := url.Parse(rpcURI); err == nil {
+					guri, _ := url.Parse(grpcURI)
+					grpcURI = uri.Hostname() + ":" + guri.Port()
+				}
+			}
+
+			grpcKeepAlive, _ := flagSet.GetDuration(flags.FlagGRPCKeepAlive)
+
+			// do not use WithReturnConnectionError() or WithBlock as we want dial happening in the background.
+			// these options makes DialContext blocking and breaks behavior of offline commands
+			// that don't set offline flag explicitly (like add-genesis-account)
+			gcl, err := grpc.DialContext(ctx, grpcURI,
+				grpc.WithInsecure(),
+				grpc.WithKeepaliveParams(keepalive.ClientParameters{
+					Time: grpcKeepAlive,
+				}),
+			)
+
+			if err == nil {
+				clientCtx = clientCtx.WithClientConn(gcl)
+			}
+		}
+	}
```

Comment:
As per this comment https://github.com/cosmos/cosmos-sdk/pull/10997#issuecomment-1019524608, let's not parse rpcURI to set grpcURI.

This should be enough:
```go
if clientCtx.ClientConn == nil || flagSet.Changed(flags.FlagGRPcNode) {
	grpcURI, _ := flagSet.GetString(flags.FlagGRPCNode)
	grpcKeepAlive, _ := flagSet.GetDuration(flags.FlagGRPCKeepAlive)

	// do not use WithReturnConnectionError() or WithBlock as we want dial happening in the background.
	// these options makes DialContext blocking and breaks behavior of offline commands
	// that don't set offline flag explicitly (like add-genesis-account)
	gcl, err := grpc.DialContext(ctx, grpcURI,
		grpc.WithInsecure(),
		grpc.WithKeepaliveParams(keepalive.ClientParameters{
			Time: grpcKeepAlive,
		}),
	)

	clientCtx = clientCtx.WithClientConn(gcl)
}
```

We also don't need to take into account `isOffline`. That's orthogonal, it's for setting sequence and account number manually using the CLI.